core/src/main/scala/org/apache/spark/ui/exec/ExecutorHeapHistogramPage.scala

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ui.exec
+
+import javax.servlet.http.HttpServletRequest
+
+import scala.xml.{Node, Text}
+
+import org.apache.spark.SparkContext
+import org.apache.spark.ui.{SparkUITab, UIUtils, WebUIPage}
+
+private[ui] class ExecutorHeapHistogramPage(
+    parent: SparkUITab,
+    sc: Option[SparkContext]) extends WebUIPage("heapHistogram") {
+
+  // Match the lines containing object informations
+  val pattern = """\s*([0-9]+):\s+([0-9]+)\s+([0-9]+)\s+(\S+)(.*)""".r
+
+  def render(request: HttpServletRequest): Seq[Node] = {
+    val executorId = Option(request.getParameter("executorId")).map { executorId =>
+      UIUtils.decodeURLParameter(executorId)
+    }.getOrElse {
+      throw new IllegalArgumentException(s"Missing executorId parameter")
+    }
+    val time = System.currentTimeMillis()
+    val maybeHeapHistogram = sc.get.getExecutorHeapHistogram(executorId)
+
+    val content = maybeHeapHistogram.map { heapHistogram =>
+      val rows = heapHistogram.map { row =>
+        row match {
+          case pattern(rank, instances, bytes, name, module) =>
+            <tr class="accordion-heading">
+              <td>{rank}</td>
+              <td>{instances}</td>
+              <td>{bytes}</td>
+              <td>{name}</td>
+              <td>{module}</td>
+            </tr>
+          case pattern(rank, instances, bytes, name) =>
+            <tr class="accordion-heading">
+              <td>{rank}</td>
+              <td>{instances}</td>
+              <td>{bytes}</td>
+              <td>{name}</td>
+              <td></td>
+            </tr>
+          case _ =>
+            // Ignore the first two lines and the last line
```

Comment:
In addition, we will ignore all irregular messages in addition to these three lines.

Reply:
So the error will go to Executor log, and in UI it just doesn't show heap dump (empty)?

Reply:
Yes, correct. We log and return `None` for `Option[Array[String]]`. So, the heap histogram UI will be empty.

https://github.com/apache/spark/blob/284029b6582436b193d52ca0e2583c5ac7e00993/core/src/main/scala/org/apache/spark/SparkContext.scala#L756-L775